### DISCUSSION TITLE
Queueing System/Secondary Storage Tier

### DISCUSSION THREAD

**Discussion by brandongalbraith** [bibanon/tubeup#298] [Ideas]
While grabbing lunch at the Internet Archive recently, tubeup came up. I was provided a feature request by the Internet Archive staff: the ability for tubeup to operate in a mode where content is retrieved, but stored locally instead of being immediately uploaded to the Internet Archive. On future runs, if a multimedia artifact was detected as no longer being available (for example, a Youtube video was deleted), that would trigger the upload to the Internet Archive.

I am still putting thought into this request, but wanted to share the high level concept for socialization amongst project users.

**Comment by vxbinaca** [comment]
I think I know who told you that because that person and I discussed it a few months ago when talking about another matter.

I don't really control Tubeup. I'm just here for a shift sort of directing things. If you explicitly need by blessing (for some odd reason), the person who I talked to at IA already got it. I'd talk to @antonizoon though.

I think you have balls for bringing up the tool there it's caused them heartburn totally unintentional by me. If I ever took a tour of IA I'd do it stealth and not mention my full name or especially the tool. I have a chance of not being recognized because theres almost no pictures of me on the internet.

Edit 2: My stance: Give IA whatever they want. The tool exists because they tolerate it. Talk to Ant.

Edit 3: Rips are about to become extinct anyway due to [odious Google web DRM.](https://www.youtube.com/watch?v=0i0Ho-x7s_U). We've already seen it in other extractors like TikTok or Instagram who aggressively prevent ripping.

**Comment by antonizoon** [comment]
This actually seems to strike the right balance between user anxiety due to the general state of the internet let alone youtube, and only uploading what is absolutely necessary because there is only so much space at the IA.

In the past the situation may have been different but nowadays users are now mass uploading ahead of any problems and this time compared to all other situations, the user for once might be right.

If your team knows exactly what to do, go ahead and write up the code for everyone to review in a pull request. We also can approve additional Github contributors as necessary.

If your team really needs some development assistance, we may have some motivated volunteers in light of the special situation, but I think for a tool that is a significant pipeline to the Internet Archive let's let you all see what you can make first.

**Comment by brandongalbraith** [comment]
Revisiting this idea [with the Internet Archive being storage constrained due to the cost of hard drives due AI investment](https://www.404media.co/the-ai-hard-drive-shortage-is-making-it-more-expensive-and-harder-to-archive-the-internet/).

My proposal in the interim (supporting other target storage systems natively with an option flag) is an option to bypass uploading the downloaded artifacts, and instead package them into a zip file for local storage. Thoughts?

**Comment by vxbinaca** [comment]
This proposal doesn't follow the unix philosophy, part of which is do one thing and do it well. The feature you want is scope creep.

There is a tool called [Tubearchivist ](https://github.com/tubearchivist/tubearchivist) consider looking into it.

**Comment by vxbinaca** [comment]
If IA is getting squeezed by storage prices, you are too. Either way, consider slowing ingestion. I have. I'm down like 70 percent. Youtube agressivly cycling cookies doesn't help.

Local storage is a feature that invites content corruption from filesystems that aren't meant for archival held on bottom tier (Seagate, Team Group) storage, or alteration of the metadata or content by the user.

I don't want this feature and Antonizoon doesn't either in the proceeding years. It's been discussed over and over.

**Reply by brandongalbraith** [reply]
No worries, will folk and implement 👍

**Reply by vxbinaca** [reply]
I think you should consider slowing your use if you're concerned about IAs costs and donate money to them.

If you fork I'm going to insist you strip out the Tubeup name out of the fork and especially the scannerID and BibAnons name too.
